src/adapter/src/subscribe.rs

```diff
@@ -154,44 +165,95 @@ impl ActiveSubscribe {
                                         .collect_vec();
 
                                     // Four cases:
-                                    // [(key, value, +1)] => ("upsert", key, value
-                                    // [(key, v1, -1), (key, v2, +1)] => ("upsert", key, v2)
-                                    // [(key, value, -1)] => ("delete", key, NULL)
-                                    // everything else => ("key_violation", key, NULL)
+                                    // [(key, value, +1)] => ("insert", key, NULL, value)
+                                    // [(key, v1, -1), (key, v2, +1)] => ("upsert", key, v1, v2)
+                                    // [(key, value, -1)] => ("delete", key, value, NULL)
+                                    // everything else => ("key_violation", key, NULL, NULL)
+                                    let value_columns = self.arity - order_by_keys.len();
                                     let mut packer = row_buf.packer();
                                     new_rows.push(match &group[..] {
-                                        [(_, row, 1)] | [(_, _, -1), (_, row, 1)] => {
-                                            // upsert
+                                        [(_, row, 1)] => {
+                                            packer.push(if debezium {
+                                                Datum::String("insert")
```

Comment:
Can these both be either consts or hard coded. Don't care which, but should be the same.

Reply:
done, all hard coded